pkg/e2e/actuators/actuators.go

```diff
@@ -0,0 +1,441 @@
+package actuators
+
```

Comment:
This is called actuators but the fact that this is driven by the actuator is circumstantial today, this could be e.g the machine controller. This should belong to the same suite there's in infra and may be rename it to "elastic compute" or similar

Reply:
I did not want to overload the word controller. It's true machine controller is "just" a wrapper around an actuator. Here, `actuator` is shorter than `machinecontroller` or `elasticcompute`. Plus, elastic compute is process oriented term compared to actuator which is component oriented. To goal was to break the tests into component-wise buckets. Infra homes all tests of which I was not sure what would be the right component to put them to. More or less following what I described in the test design google doc.

Reply:
the suite you are reimplementing belongs all together. It's validating holistic features that result from the interaction of one or more of the components that provide the e2e machine API experience. Actuator is an implementation detail that we don't care about in this context cc 
We should have only one suite pkg e.g "elastic compute", may be different files and a common utils
@bison @frobware

Reply:
I tend to agree that these are meant to be high-level E2E, so things should be named more after groups of features / use-cases / stories than components. Not really sure what the right thing is exactly though. Any example in upstream Kubernetes or something we can look to?

Reply:
https://github.com/kubernetes/kubernetes/tree/master/test/e2e

Reply:
> We should have only one suite pkg e.g "elastic compute", may be different files and a common utils

There is only one test suite. Under the hood broken into more packages for better readability and manipulation.

1. To be clear the conflict of interest is to decide if we are going to break the single test suite component-wise or process-wise. The test suite for me here is "Conformance". We might even go with `pkg/e2e/conformance/...` layout to explicitly say the tests fall under the category (though I don't consider that necessary).

   I am fine with both (component and process). Any hints on what would be the names/identifiers we might use for holistic process-wise categorization? I could imagine:
   - `autoscaling`
   - `machinereconciling`
   - anything else?

   `elasticcompute` is too generic buzzword that does not say anything.

2. Does the conflict also includes the question of putting the tests under multiple files under one package instead of how it is currently done?

   I still want to break the individual tests into packages so you can import subset of tests in different test suite (e.g. `Performance`). If you check the import part of `e2e_test.go`, you can see:

   ```go
   import (
   	...
   	_ "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/actuators"
   	_ "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/autoscaler"
   	_ "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/infra"
   	_ "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/operators"
   )
   ```

   I could have `pkg/e2e/performance/e2e_test.go` that will import just operators (to assert both operators are available and working). With all tests under the same package (just different files) you can import either all the tests or no tests. And I don't want to import all conformance tests into the performance test suite.

   And again, it's easier to see intention of a set of tests if they are imported from e.g. `actuators` or `autocaler` then to just have:
   ```go
   import (
   	...
   	_ "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e"
   )
   ```

---
*Bottom line*: On the other hand, one could argue we can put all the tests under one huge test suite and use `[Conformance]` and `[Performance]`. Doing that, we would need to put the keyword to every `Describe` or spec part to make sure only conformance, resp. performance tests are run when `--ginkgo.focus` is set to `[Conformance]`, resp. `[Performance]` focus. Though, this can be discussed later once we have more test suites.

Reply:
1 - This must be process/feature wise, it does not care about components. Conformance/machines/smoke/elasticCompute I don't have strong opinions on the naming

2 - "you can import either all the tests or no tests." this is good and it's exactly the invariant we want to force. This suite is the gate for any of repo which might have an impact on the user feature validation

Reply:
> 1 - This must be process/feature wise, it does not care about components. Conformance/machines/smoke/elasticCompute I don't have strong opinions on the naming

`Conformance` is the test suite name, `machines`, `autoscaling`, ... would be test suite groups. The question is what would be the group names?

> 2 - "you can import either all the tests or no tests." this is good and it's exactly the invariant we want to force. This suite is the gate for any of repo which might have an impact on the user feature validation

This is orthogonal to the question. All tests are to be run as part of `Conformance` test suite. Inseparably from each other. The intent is to:
- allow other test suites to reuse subset of tests
- decompose one huge test suite into smaller chunks (groups) so it's easier to maintain, read and extend the code